tests/test_build_html.py

```diff
@@ -662,7 +660,7 @@ def test_numfig_without_numbered_toctree(app, status, warning):
             yield check_xpath, etree, fname, xpath, check, be_found
 
 
-@gen_with_app(buildername='html', testroot='numfig',
+@gen_with_app(buildername='html', testroot='numfig', srcdir='test_build_html_numfig_on',
```

Comment:
If I remove gen_with_app, this PR became as same as #3300.
It seems that you have fully understood the pytest code, so I think it's time to review #3300.